File: app/src/main/java/com/m7mdabaza/fooddelivery/ui/activities/HomeActivity.java
package com.m7mdabaza.fooddelivery.ui.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.m7mdabaza.fooddelivery.R;
import com.m7mdabaza.fooddelivery.ui.fragments.CartFragment;
import com.m7mdabaza.fooddelivery.ui.fragments.FavouriteFragment;
import com.m7mdabaza.fooddelivery.ui.fragments.HomeFragment;
import com.m7mdabaza.fooddelivery.ui.fragments.ProfileFragment;

public class HomeActivity extends AppCompatActivity {

    private BottomNavigationView bottomNavigation;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        findViewById();

        bottomNavigation.setOnNavigationItemSelectedListener(navigationItemSelectedListener);
        openFragment(new HomeFragment());
        if (null == savedInstanceState) {

            getSupportFragmentManager().beginTransaction().replace(R.id.fragmentContainerView, new HomeFragment()).commit();
        }
    }

    BottomNavigationView.OnNavigationItemSelectedListener navigationItemSelectedListener =
            item -> {
                switch (item.getItemId()) {
                    case R.id.navigation_home:
                        openFragment(new HomeFragment());
                        break;
                    case R.id.navigation_favourite:
                        openFragment(new FavouriteFragment());
                        break;
                    case R.id.navigation_cart:
                        openFragment(new CartFragment());
                        break;
                    case R.id.navigation_profile:
                        openFragment(new ProfileFragment());
                        break;
                }

                return true;
            };


    public void openFragment(Fragment fragment) {
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.fragmentContainerView, fragment);
        transaction.addToBackStack(null);
        transaction.commit();
    }

    void findViewById() {
        //BottomNavigationView
        bottomNavigation = findViewById(R.id.bottom_navigation);

    }
}

File: app/src/main/java/com/m7mdabaza/fooddelivery/ui/activities/HomeActivity.kt
package com.m7mdabaza.fooddelivery.ui.activities

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.m7mdabaza.fooddelivery.R
import com.m7mdabaza.fooddelivery.ui.fragments.*

import kotlinx.android.synthetic.main.activity_home.*


class HomeActivity : AppCompatActivity() {



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        bottomNavigation.setOnNavigationItemSelectedListener(navigationItemSelectedListener)
        openFragment(HomeFragment())
        if (null == savedInstanceState) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragmentContainerView, HomeFragment()).commit()
        }

    }

    private var navigationItemSelectedListener =
        BottomNavigationView.OnNavigationItemSelectedListener { item: MenuItem ->
            when (item.itemId) {
                R.id.navigation_home -> openFragment(HomeFragment())
                R.id.navigation_favourite -> openFragment(FavouriteFragment())
                R.id.navigation_cart -> openFragment(CartFragment())
                R.id.navigation_profile -> openFragment(ProfileFragment())
            }
            true
        }

    private fun openFragment(fragment: Fragment) {
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.fragmentContainerView, fragment)
        transaction.addToBackStack(null)
        transaction.commit()
    }

}